code/model/entity/abstract.php

```diff
@@ -269,10 +269,14 @@ public function setProperty($name, $value, $modified = true)
      */
     public function hasProperty($name)
     {
+        if (empty($name)) {
+            return false;
```

Comment:
Sure, just saying this stuff gets complex when nesting if statements or switching on multiple conditions.